lib/file.h

```diff
@@ -59,5 +61,6 @@ int         safe_fsync_close(FILE **f);
 int         safe_open(const char *path, int flags);
 int         safe_rename(const char *src, const char *target);
 int         safe_symlink(const char *oldpath, const char *newpath);
+int         to_absolute_path(char *path, const char *reference);
```

Comment:
If there's only one consumer of this function, the need to move it to file.c is discutable. But if you choose to do so, a name like file_abspath or mutt_abspath (the header seems schizophrenic on the prefix that should be used) would definitely be a plus.

Reply:
> If there's only one consumer of this function

There are a handful of one-off functions like this.
I've chosen to move them, for now, so they're more discoverable.
The more comparisons we can make between, say file functions, the easier it should be to refactor and improve.